statesman-engine/src/main/java/io/appform/statesman/engine/MessageConfigStore.java

```diff
@@ -0,0 +1,13 @@
+package io.appform.statesman.engine;
+
+import io.appform.statesman.model.MessageConfig;
+import io.appform.statesman.model.action.template.ActionTemplate;
+
+import java.util.Optional;
+
+public interface MessageConfigStore {
+
+    Optional<MessageConfig> create(MessageConfig messageConfig);
```

Comment:
should create return optional ? if it fails, it should throw the exception right rather than fail internally ?